Remove SAML IDP feature

hsinn0 opened the pull request:
- In preparation for the SAML library replacement

~~Approve or disapprove but do not merge to develop yet.~~ We want merge this to `develop` now. As this is a breaking change, ~~we have to decide which UAA version this change should go with.~~ we also want to bump UAA major version after the merge.

swalchemist committed:
delete: saml/idp directory & code

* Comment out / delete code to get the project to work
* status: all unit tests passing (./gradlew test), local server starts (./gradlew run) successfully with the UAA login page working in browser

[#182118433]

Co-authored-by: Alicia Yingling <alicia.yingling@broadcom.com>
Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>
Co-authored-by: Danny Faught <danny.faught@broadcom.com>
Co-authored-by: Peter Chen <peter-h.chen@broadcom.com>

hsinn0 committed:
delete: SAML IDP tests

[#182118433]

Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>

hsinn0 committed:
delete: SAML-IdP-related content in UAA API doc

[#182118433]

Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>

hsinn0 committed:
delete: /saml/service-providers endpoint code

[#182118433]

Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>

hsinn0 committed:
delete: More references to /saml/service-providers endpoint

[#182118433]

Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>

hsinn0 committed:
fix: `generateDocs` gradle task failure

- Changed the doc test to use mocked-up data instead of making actual calls to the end-point that does not exist anymore.

[#182118433]

Co-authored-by: Alicia Yingling <alicia.yingling@broadcom.com>

Tallicia committed:
fix/wip: remove test for idp endpoint spIsAuthenticated is broken

- The unit test was testing the idp endpoint which is removed, so the test is now removed.

[#182118433]

Co-authored-by: Hongchol Sinn <hongchol.sinn@broadcom.com>

hsinn0 committed:
fix: unit test failures

- Disabled/removed test cases that depend on the removed SAML IDP feature.

[#182118433]

Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>

hsinn0 committed:
fix: Token endpoint doc test

- Replaced the use of the removed SAML IDP endpoint call with mockup
  data.
- Also, some clean-ups and refactoring

[#182118433]

Co-authored-by: Hongchol Sinn <hongchol.sinn@broadcom.com>
Co-authored-by: Danny Faught <danny.faught@broadcom.com>
Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>
Co-authored-by: Alicia Yingling <alicia.yingling@broadcom.com>

hsinn0 committed:
Replace SAML2 bearer grant integration test cases with a mock mvc test.

- The integration test was getting the IDP metadata from actual UAA server, which needs be bypassed as UAA does not have the SAML IDP endpoint anymore.
- With that, the existing integration test basically becomes the same as `TokenEndpointDocs.getTokenUsingSaml2BearerGrant()`.
- So removed the existing integration test cases and created a new mock mvc test based on the `TokenEndpointDocs` code.

[#182118433]

Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>

hsinn0 committed:
remove UAA-as-SAML-IDP related tests

- These inputs to the parameterized tests are testing "/saml/idp/metadata" endpoint
which has been removed

[#182118433]

Co-authored-by: Peter Chen <peter-h.chen@broadcom.com>

hsinn0 committed:
remove duplicate tests

- the logic being tested in these two tests are
already covered by Saml2BearerGrantMockMvcTests
- since UAA-as-a-SAML-IDP feature has been removed, the
difference between these two tests (same zone vs two zones) do not
matter so they can both be covered by Saml2BearerGrantMockMvcTests

[#182118433]

Co-authored-by: Peter Chen <peter-h.chen@broadcom.com>

hsinn0 committed:
remove unnecessary test setup

- this 3 lines have no bearing on these 2 tests

Co-authored-by: Peter Chen <peter-h.chen@broadcom.com>

bruce-ricard committed:
refactor: rename var

- to clarify things
- and correct wrong comments

[#182118433]

Co-authored-by: Peter Chen <peter-h.chen@broadcom.com>

peterhaochen47 committed:
refactor: maintain test coverage

- the replaced MockMvc test does not work anyway because
it relies on UAA's ability to function as a SAML IDP, which
has been removed in prior commits
- replace it with an integration test (which instead uses
a deployed test SAML IDP server) to maintain test coverage
with the following minor differences:
  - the new test uses auth code grant flow instead of the
  password grant in the old test. We think this detail should
  not affect the key test intention (testing whether SAML data
  is correctly translated into OAuth data).
  - the old test also tests that the refresh token flow, which
  we do not think affect the key test intention either. Hence,
  we omit this in the new test.

[#182118433]

Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>

peterhaochen47 committed:
delete test related to the removed feature UAA-as-SAML-IDP test

- the UAA-as-SAML-IDP feature has been removed
- this test exclusively about UAA-as-SAML-IDP because it tests
the output of the /saml/idp/initiate endpoint. The linked story
in the original commit that added this test
(https://github.com/cloudfoundry/uaa/commit/bee262d44e53a22d3fbaf74e9a0d2895cf201542)
also indicates such.

[#182118433]

Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>

peterhaochen47 committed:
delete unnecessary test

- this test relies on UAA's ability to function as
a SAML IDP, which is a feature that has been removed
- we opt to simply remove this test because its tested
behavior has been sufficiently covered by existing tests:
  - this test simulates a UAA login via an external SAML
  IDP, where UAA consumes a SAML assertion. This has been
  extensively tested on the integration level in SamlLoginIT (https://github.com/cloudfoundry/uaa/blob/4490877e016d6491d02c8b256fe8e8e4986afc8e/uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/SamlLoginIT.java)
  - this test verifies some loggings, which I think have been
  sufficiently covered by the combination of the AuditCheckMockMvcTests
  (https://github.com/cloudfoundry/uaa/blob/4490877e016d6491d02c8b256fe8e8e4986afc8e/uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/audit/AuditCheckMockMvcTests.java)
  which tests that UAA Event will be logged once user scim endpoint is
  triggered, and the LoginSamlAuthenticationProviderTests (https://github.com/cloudfoundry/uaa/blob/4490877e016d6491d02c8b256fe8e8e4986afc8e/server/src/test/java/org/cloudfoundry/identity/uaa/provider/saml/LoginSamlAuthenticationProviderTests.java#L584) which checks that the user scim endpoint is
  triggered upon a new SAML external login.
- also: clean up unused methods, vars, and imports in this class

[#182118433]

Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>

hsinn0 committed:
delete: commented-out code

[#182118433]

Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>

hsinn0 committed:
refactor: Move bean definition

- Moved a bean definition from `saml-idp.xml`, which is being deleted, to `saml-providers.xml`.

[#182118433]

Co-authored-by: Bruce Ricard <bruce.ricard@broadcom.com>

strehle reviewed (commented):
Check and change Readme notice but the rest LGTM

strehle commented:
As mentioned in slack , please wait until release 76.31.0, then merge this PR and increase major version, thanks

hsinn0 committed:
doc: Update README for the SAML IDP feature removal

[#182118433]

swalchemist commented on README.md:
```diff
@@ -339,3 +337,8 @@ To debug UAA and LDAP integrations, we use an OpenLdap docker image from [VMWare
 Use below command to clean-up container and volume:
 - `docker-compose down --volumes`
+
+# UAA's functionality as a SAML identity provider (IdP)
```
I don't think we need to document in the README a feature that is no longer present. That's what the release notes are for. Removing the deprecation notice is a sufficient change.

hsinn0 replied:
Removed the added section.

hsinn0 committed:
doc: Further update of README for the SAML IDP feature removal

- Remove the added section about the removed feature as that should be just mentioned in release notes.

[#182118433]

hsinn0 commented:
> Check and change Readme notice but the rest LGTM

Done
